scheduler/replica_scheduler.go

```diff
@@ -27,16 +27,12 @@ func NewReplicaScheduler(ds *datastore.DataStore) *ReplicaScheduler {
 	return rcScheduler
 }
 
-func (rcs *ReplicaScheduler) ScheduleReplica(replica *longhorn.Replica) (*longhorn.Replica, error) {
+func (rcs *ReplicaScheduler) ScheduleReplica(replica *longhorn.Replica, replicas map[string]*longhorn.Replica) (*longhorn.Replica, error) {
 	// only called when replica is starting for the first time
 	if replica.Spec.NodeID != "" {
 		return nil, fmt.Errorf("BUG: Replica %v has been scheduled to node %v", replica.Name, replica.Spec.NodeID)
 	}
-	// get replica list without current replica
-	replicas, err := rcs.ds.GetVolumeReplicas(replica.Spec.VolumeName)
-	if err != nil {
-		return nil, err
-	}
+	// delete current replica for scheduler
```

Comment:
When will this replica exist in the replicas? It seems we haven't written it into the datastore yet.

Reply:
Yes, the replica won't be written into the datastore in scheduler. It will be written into datastore  after scheduling(volume controller will save it into datastore). To solve the race condition in Scheduler, I thought it's better to avoid adding creation and list logic into scheduler.